packages/tpetra/core/src/Tpetra_CrsMatrix_def.hpp

```diff
@@ -8445,97 +8445,68 @@ CrsMatrix<Scalar, LocalOrdinal, GlobalOrdinal, Node>::
     }
     }
 
-    /*********************************************************************/
     /**** 3) Copy all of the Same/Permute/Remote data into CSR_arrays ****/
     /*********************************************************************/
 
     // Backwards compatibility measure.  We'll use this again below.
-#ifdef HAVE_TPETRA_MMM_TIMINGS
-    RCP<TimeMonitor> tmCopySPRdata = rcp(new TimeMonitor(*TimeMonitor::getNewTimer(prefix + std::string("TAFC unpack-count-resize"))));
-#endif
-    destMat->numImportPacketsPerLID_.sync_host ();
-    Teuchos::ArrayView<const size_t> numImportPacketsPerLID =
-      getArrayViewFromDualView (destMat->numImportPacketsPerLID_);
-    destMat->imports_.sync_host ();
-    Teuchos::ArrayView<const char> hostImports =
-      getArrayViewFromDualView (destMat->imports_);
 
-    if (verbose) {
-      std::ostringstream os;
-      os << *verbosePrefix << "Calling unpackAndCombineWithOwningPIDsCount"
-         << std::endl;
-      std::cerr << os.str ();
-    }
-    size_t mynnz =
-      unpackAndCombineWithOwningPIDsCount (*this,
-                                           RemoteLIDs,
-                                           hostImports,
-                                           numImportPacketsPerLID,
-                                           constantNumPackets,
-                                           INSERT,
-                                           NumSameIDs,
-                                           PermuteToLIDs,
-                                           PermuteFromLIDs);
-    if (verbose) {
-      std::ostringstream os;
-      os << *verbosePrefix << "unpackAndCombineWithOwningPIDsCount returned "
-         << mynnz << std::endl;
-      std::cerr << os.str ();
-    }
-    size_t N = BaseRowMap->getLocalNumElements ();
+    // TODO JHU Need to track down why numImportPacketsPerLID_ has not been corrently marked as modified on host (which it has been)
+    // TODO JHU somewhere above, e.g., call to Distor.doPostsAndWaits().
+    // TODO JHU This only becomes apparent as we begin to convert TAFC to run on device.
+    destMat->numImportPacketsPerLID_.modify_host(); //FIXME
```

Comment:
`numImportPacketsPerLID_` was modified last on host.  Without this, the unit tests fail.  I hadn't noticed that exception test until you just pointed it out.

Reply:
In fact, I'm guessing that check is no longer necessary, since the opposite of what it's checking is in fact true.

Reply:
Removed exception test on line 8471 below, because numImportPacketsPerLID_ *has* been modified on host somewhere above (as comment says) and then synchronized to device.